src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -298,6 +301,17 @@ def validate_user(namespace):
             "--user can only be clusterUser or clusterMonitoringUser")
 
 
+def validate_pod_ip_allocation_mode(namespace):
```

Comment:
imo validation inside the cli isn't necessary. this is what RP is for and it will make it more work to add new variants in the future.

Reply:
The original intention of verification on the cli/client side is to achieve fail fast and avoid submitting too many bad requests to rp.

Reply:
I think checking for only the permissible values in the CLI to fail fast is always helpful. Keeping this small validation as is.